src/main/java/com/google/devtools/build/lib/runtime/AuthHeadersProvider.java

```diff
@@ -0,0 +1,37 @@
+package com.google.devtools.build.lib.runtime;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Generic interface to provide authentication headers for http/grpc requests
+ * to bazel modules.
+ */
+public interface AuthHeadersProvider {
+
+  /**
+   * Returns the type of authentication mechanism used i.e. oauth.
+   */
+  String getType();
+
+  /**
+   * Returns request headers necessary for authentication to be added
+   * to the http/grpc request.
+   */
+  Map<String, List<String>> getRequestHeaders(URI uri) throws IOException;
+
+  /**
+   * Refreshes the authentication credentials.
```

Comment:
maybe say more about what this does, and when clients should call it?

Reply:
done.